main.js

```diff
@@ -16,3 +18,32 @@ navItems.forEach((link) => {
     menu.classList.toggle('hidden');
   });
 });
+function openModal(modal) {
+  if (modal == null) return;
+  modal.classList.add('active');
+  overlay.classList.add('active');
+}
+function closeModal(modal) {
+  if (modal == null) return;
+  modal.classList.remove('active');
+  overlay.classList.remove('active');
+}
+
+overlay.addEventListener('click', () => {
+  const modals = document.querySelectorAll('.modal.active');
+  modals.forEach((modal) => {
+    closeModal(modal);
+  });
+});
```

Comment:
- [ ] Firlty, based on the project requirements given,  your card details should be organized in an array containing objects, one object for each project. then dynamically use that array to render your cards. 
here's a sample :
```
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```